### DISCUSSION TITLE
How to generate coq sail library files?

### DISCUSSION THREAD

**Discussion by nazar-codethink** [rems-project/sail#352] [Q&A] [answered]
Hello,

I'm trying to learn how to use sail generated coq code, but can't find docs about generating lib files. So far I've generated coq files for duopod model and it has imports like "Require Import Sail.Base." I assume I need to generate "Sail.Base" from some another file? How can I do this?

**Comment by bacam** [comment, marked as answer]
The library files are in the https://github.com/rems-project/coq-sail repository.  It's also available as an opam package in the Coq opam repository.

**Reply by nazar-codethink** [reply]
@bacam Thanks, that works!
